### ISSUE TITLE
Autocompletion for objects don't work in debugging mode

### ISSUE THREAD

**Issue by strboul** [randy3k/radian#194]
The auto-completion feature for the objects doesn't seem to work when the radian gets into the debugging mode. This looks like a bug. An example to reproduce it:

```r
R>> square <- function(x) {
  browser()
  apply(x, 2, mean)
}
R>> square(mtcars)
Called from: square(mtcars)
Browse[1]> ls()
[1] "x"
Browse[1]> x$  # <--- doesn't autocomplete
Browse[2]> Q
R>> df <- mtcars
R>> df$  # <--- autocomplete works
```

Thanks a ton again! I love radian :heart:

**Comment by randy3k** [comment]
IIRC, the plain R console also has the same issue? So it h it s an upstream issue!? (On mobile now)

**Comment by strboul** [comment]
That's correct. I haven't realized that it is also an issue on the plain R console.

**Comment by randy3k** [comment]
It seems that RStudio has been something with it to make it work.

**Comment by strboul** [comment]
I can't seem to find anything on [bugzilla](https://bugs.r-project.org/bugzilla/). Maybe hasn't been reported yet. Strange...

**Comment by randy3k** [comment]
[It is](https://github.com/rstudio/rstudio/blob/efe75a882289178d7f8bd2525e3bf4ec9c316ea7/src/cpp/session/modules/SessionRCompletions.R#L1948) how rstudio handles debugger completion, we might be able to borrow something there. Or we could rewrite a completion engine, in that case also fixes https://github.com/randy3k/radian/issues/29

**Comment by strboul** [comment]
> [It is](https://github.com/rstudio/rstudio/blob/efe75a882289178d7f8bd2525e3bf4ec9c316ea7/src/cpp/session/modules/SessionRCompletions.R#L1948) how rstudio handles debugger completion, we might be able to borrow something there. Or we could rewrite a completion engine, in that case also fixes #29

Great idea, for sure. I also believe it's a quite a big task.

I am curious if it's possible to use LSP for the radian completion instead of relying on the R console. Seems that LSP is much more flexible.

**Comment by lafet1** [comment]
Hi, I just recently started using `radian` and stumbled on this issue. I'm curious whether there's been any development on this front or not. Thanks!
